cli/cmd/inject.go

```diff
@@ -149,7 +160,7 @@ func (rt resourceTransformerInject) transform(bytes []byte) ([]byte, []inject.Re
 		return nil, nil, err
 	}
 
-	if conf.IsControlPlaneComponent() && !rt.injectProxy {
+	if conf.IsControlPlaneComponent() && rt.injectProxy == injectProxyAnnotation {
```

Comment:
Even in the `noop` case we wouldn't want to trigger that error. But that led me to check what we do in that case. And we're actually doing nothing. Even the report is being discarded. So that led me to revise this approach in #3784 which is much simpler. LMKWYT.